Data/Scripts/001_Technical/002_RubyUtilities.rb

```diff
@@ -162,6 +159,145 @@ def self.move(source, destination)
+  def to_i
+    return self.alpha.to_i << 24 | self.blue.to_i << 16 | self.green.to_i << 8 | self.red.to_i
+  end
+  #-----------------------------------------------------------------------------
+  # return color from byte integer value
+  #-----------------------------------------------------------------------------
+  def self.from_i(color)
+    b =  color & 255
+    g = (color >> 8) & 255
+    r = (color >> 16) & 255
+    a = (color >> 24) & 255
+    return Color.new(r, g, b, a)
+  end
+  #-----------------------------------------------------------------------------
+  # parse color input to return color object
+  #-----------------------------------------------------------------------------
+  def self.parse(color)
+    if color.is_a?(Color) # color object
+      return color
+    elsif color.is_a?(String) # string
+      if color.include?("#") # hex color
+        return Color.new(color)
+      elsif color.include(",") # RGB color
+        rgb = color.split(",").map(&:to_i)
+        return Color.new(*rgb)
+      end
+    elsif color.is_a?(Numeric) # decimal color
+      return Color.new(color)
+    end
+    # returns nothing if wrong input
+    return nil
+  end
+  #-----------------------------------------------------------------------------
+  # returns color object for some commonly used colors
+  #-----------------------------------------------------------------------------
+  def self.red; return Color.new(255, 0, 0); end
+  def self.green; return Color.new(0, 255, 0); end
+  def self.blue; return Color.new(0, 0, 255); end
+  def self.black; return Color.new(0, 0, 0); end
+  def self.white; return Color.new(255, 255, 255); end
+  def self.yellow; return Color.new(255, 255, 0); end
+  def self.orange; return Color.new(255, 155, 0); end
+  def self.purple; return Color.new(155, 0, 255); end
+  def self.brown; return Color.new(112, 72, 32); end
+  def self.teal; return Color.new(0, 255, 255); end
+  def self.magenta; return Color.new(255, 0, 255); end
+  #-----------------------------------------------------------------------------
+end
+
+#===============================================================================
+#  Wrap code blocks in class which passes data accessible as instance
+#  variables within the code block
+#
+#  wrapper = CallbackWrapper.new { puts @test }
+#  wrapper.set(test: 'Hi')
+#  wrapper.execute  #=>  'Hi'
+#===============================================================================
+class CallbackWrapper
+  @params = {}
+  #-----------------------------------------------------------------------------
+  #  constructor
+  #-----------------------------------------------------------------------------
+  def initialize(&block)
+    @code_block = block
+  end
+  #-----------------------------------------------------------------------------
+  #  execute callback
+  #-----------------------------------------------------------------------------
+  def execute(block = @code_block, *args)
```

Comment:
I don't think `block = @code_block` should be in the parameters. If you want to use `@code_block` and also other args, well, you can't; you can't access `@code_block` from outside, and using `callbacks.execute(nil, arg1, arg2)` will just set `block` to nil. At least, that's my understanding of it.

Reply:
True true, I haven't yet used a scenario with the args + code block def on initialize ... but yeah, it wouldn't really work in this case.